Formula/c/chatgpt-cli.rb

```diff
@@ -0,0 +1,33 @@
+class ChatgptCli < Formula
+  desc "CLI for interacting with the OpenAI ChatGPT API"
+  homepage "https://github.com/kardolus/chatgpt-cli"
+  url "https://github.com/kardolus/chatgpt-cli/archive/refs/tags/v1.5.5.tar.gz"
+  sha256 "158620e871411053310c6882a3e164e013389b0b6d99dd6fb42d8d82c988e51d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    git_commit = "422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0"
```

Comment:
As with your previous iteration fo this PR, we are unable to utilise a static variable here as the formula should be able to utilise the `brew bump` command without additional intervention (in this case, manually changing this value).

See this search to see how other `go` formulae handle this issue.
https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core%20main.gitCommit&type=code

Reply:
Thanks @bevanjkay, I proposed multiple alternatives but did not get an answer. I assume that most of my users will still install the CLI outside of brew. A version and commit has is not a static combination. Folks that install my CLI outside of brew may have a different hash/version combo (they may be compiling from a fork or from edge). 

If I go with your change this is what the output of my version command will be. This is a breaking change:
```
 2024-04-21 09:04:51 ⌚  Guillermos-MacBook-Pro in ~/workspace/homebrew-core
± |master U:1 ✗| → chatgpt --version
ChatGPT CLI version v1.5.5 (commit local)
```
The current output looks like this: 
```
 2024-04-21 08:38:59 ⌚  Guillermos-MacBook-Pro in ~/workspace/homebrew-core
± |master U:1 ✗| → chatgpt --version
ChatGPT CLI version 1.5.5 (commit 422f3b9dc0b290ed7b7bc86d1d6c83ff79266de0)
```
I do not know how folks are consuming the CLI downstream. Getting rid of the hash for all my users would mean a breaking change. Thus homebrew forcing me to go to the `1.6` line instead of `1.5`. 

I proposed a change in which my non brew users (ie folks at Stripe and Shopify) will still see the git hash. However, folks that install my CLI from homebrew would see something like this: 
```
 2024-04-21 09:04:51 ⌚  Guillermos-MacBook-Pro in ~/workspace/homebrew-core
± |master U:1 ✗| → chatgpt --version
ChatGPT CLI version v1.5.5 (homebrew)
```
I can make this change but I will have to bump the version to `v.1.5.6`. I was wondering if this is something that works for y'all. I think `ChatGPT CLI version v1.5.5 (commit local)` is not very user friendly. Do you think it is?

Reply:
I can also just open up a PR in 6 months or so if that's better. I understand that you all must be busy to think with me on a solution. I can just remove the git commit for homebrew users, keep it for the rest. Then leave the variable out and in that way y'all won't notice the git-commit for non-brew users (and it won't be part of the formula).

Reply:
What if we just set it to `homebrew`? You'll know that it was built here if you get a report, and it can just be cleaned up whenever you push your next release?
`ChatGPT CLI version v1.5.5 (commit homebrew)`

Reply:
That works. Thanks @bevanjkay. I will make the change.

Reply:
we should not allow any commit hash in new go formulae.